src/bots/liquidator.ts

```diff
@@ -396,14 +393,54 @@ export class LiquidatorBot implements Bot {
 			}
 		}
 
+		if (this.isTwapping) {
+			const nowSec = Math.floor(Date.now() / 1000);
+			this.twapExecutionProgresses = new Map<string, TwapExecutionProgress>();
+			for (const marketIndex of this.perpMarketIndicies) {
+				const subaccount = this.perpMarketToSubaccount.get(marketIndex);
+				this.twapExecutionProgresses.set(
+					this.getTwapProgressKey(MarketType.PERP, subaccount, marketIndex),
+					new TwapExecutionProgress({
+						currentPosition: new BN(0),
+						targetPosition: new BN(0), // target positions to close
+						overallDurationSec: this.liquidatorConfig.twapDurationSec,
+						startTimeSec: nowSec,
+					})
+				);
+			}
+
+			for (const marketIndex of this.spotMarketIndicies) {
+				const subaccount = this.spotMarketToSubaccount.get(marketIndex);
+				this.twapExecutionProgresses.set(
+					this.getTwapProgressKey(MarketType.SPOT, subaccount, marketIndex),
+					new TwapExecutionProgress({
+						currentPosition: new BN(0),
+						targetPosition: new BN(0), // target positions to close
+						overallDurationSec: this.liquidatorConfig.twapDurationSec,
+						startTimeSec: nowSec,
+					})
+				);
+			}
+		}
+
 		// jupiter only works with mainnet
 		if (config.useJupiter && this.runtimeSpecs.driftEnv === 'mainnet-beta') {
 			this.jupiterClient = new JupiterClient({
 				connection: this.driftClient.connection,
 			});
 		}
+	}
 
-		this.liquidatorConfig = config;
+	private isTwapping() {
```

Comment:
nitpick: when i read this, thought it would mean liquidator is currently twapping, instead of that it should use twap. could maybe make it "useTwap" or "shouldTwap"